### ISSUE TITLE
Space velocity Errors

### ISSUE THREAD

**Issue by sivdur** [astropy/astropy#14405]
### What is the problem this feature will solve?

I use the following code to calculate space velocities respect to the LSR. What I want to calculate is their uncertainties. 
```
clm = SkyCoord(ra=rave_ra*u.deg,
               dec=rave_dec*u.deg,
               distance=rave_dist*u.kpc,
               pm_ra_cosdec=rave_pmra*u.mas/u.yr,
               pm_dec=rave_pmdec*u.mas/u.yr,
               radial_velocity=rave_RV*u.km/u.s,
               frame='icrs'
              )

gclm = clm.transform_to(coord.Galactocentric(galcen_v_sun=[8.5, 13.38, 6.49]*u.km/u.s))

cUlsr = gclm.velocity.d_x
cVlsr = gclm.velocity.d_y
cWlsr = gclm.velocity.d_z
```

### Describe the desired outcome

I want to calculate space velocity errors by giving the proper motion, distance, and radial velocity errors. But neither SkyCoord nor ICRS don't accept errors of pm, distance, radial velocity as inputs.

**Comment by mhvk** [comment]
Yes, that would be nice! There is a long-standing wish to allow quantities to have uncertainties, but this is still not implemented (see #3715 for an aborted attempt). In principle, it should be possible to use the `Distribution` class, which would also nicely give you a sense of non-linear dependence on input uncertainties. But I haven't tried to see whether those actually work completely properly with `SkyCoord` yet -- again, something hopefully to be done at some point (maybe by you if it doesn't work?!).

Still, even if `Distribution` does not work, the idea is certainly OK: generate inputs with distributions given by their errors, convert, and infer the output uncertainties from the results.

**Comment by sivdur** [comment]
@mhvk actually Galyp can propagate distance, radial velocity, and proper motion uncertainties to Galactic coordinates. Here is the [link](https://docs.galpy.org/en/v1.8.1/reference/coordscovdvrpmllbbtovxyz.html).
Maybe it can be implemented to astropy.

**Comment by mhvk** [comment]
@sivdur - thanks for the link. It makes sense that a specialized package would have this. For astropy, I think we'd ideally have a more general mechanism, that works for all transformations.  A helper function under `uncertainty` might well be something to consider. Though I'd like first to integrate `Distribution` better (and perhaps have this `Variable` class of #3715...)
